dimod/higherorder.py

```diff
@@ -23,8 +23,92 @@
 from dimod.binary_quadratic_model import BinaryQuadraticModel
 from dimod.decorators import vartype_argument
 from dimod.vartypes import Vartype
+import dimod
+from dimod.response import SampleSet
+import numpy as np
 
-__all__ = ['make_quadratic']
+__all__ = ['make_quadratic','HigherOrderComposite']
+
+
+class HigherOrderComposite(dimod.ComposedSampler):
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['penalty_strength'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, h, J, offset=0, penalty_strength=1.0, **parameters):
+        bqm = BinaryQuadraticModel(linear=h, quadratic={}, offset=0,
+                                   vartype=dimod.SPIN)
+        bqm = make_quadratic(J, penalty_strength, bqm=bqm)
+        response = self.child.sample(bqm, **parameters)
+
+        return polymorph_response(response, h, J, offset, bqm,
+                                  penalty_strength=penalty_strength)
+
+
+def polymorph_response(response, h, J, offset, bqm, penalty_strength=None):
+    record = response.record
+    original_variables = set(
+        sorted(set(h.keys()) | set(v for key in J.keys() for
```

Comment:
In python3 you can't sort unlike types so this will fail for `h = {'a': -1, 0: 1}`